hasktorch/src/Torch/Typed/Functional.hs

```diff
@@ -6063,20 +6173,60 @@ upsample_bicubic2d _align_corners _input = unsafePerformIO $ (ATen.cast3 ATen.Ma
 -- upsample_nearest1d :: Tensor device dtype shape -> Int -> Tensor device dtype shape
 -- upsample_nearest1d _input _output_size = unsafePerformIO $ (ATen.cast2 ATen.Managed.upsample_nearest1d_tl) _input _output_size
 
--- | Applies a 2D bicubic upsampling to an input signal composed of several input channels.
+-- | Applies a 2D nearest neighbor upsampling to an input signal composed of several input channels.
 --
 -- >>> (dtype &&& shape) $ upsample_nearest2d @3 @5 (ones :: CPUTensor 'D.Float '[2,3,2,2])
 -- (Float,[2,3,3,5])
+
 upsample_nearest2d ::
-  forall w h shape dtype device.
+  forall h w shape dtype device.
   (KnownNat h, KnownNat w, All KnownNat shape) =>
   Tensor device dtype shape ->
   Tensor device dtype (Upsample2d shape h w)
-upsample_nearest2d _input = unsafePerformIO $ (ATen.cast2 ATen.Managed.upsample_nearest2d_tl) _input ([w, h] :: [Int])
+upsample_nearest2d _input = unsafePerformIO $ (ATen.cast2 ATen.Managed.upsample_nearest2d_tl) _input ([h, w] :: [Int])
   where
     w = natValI @w :: Int
     h = natValI @h :: Int
 
+
+-- Freeform Resizing
+interpolate2d :: 
+  forall newShape mode w h shape dtype device. 
+  (KnownShape shape, KnownNat w, KnownNat h, KnownSymbol mode) => 
+  Bool -> 
+  Tensor device 'D.Float shape -> 
+  Tensor device 'D.Float newShape
+interpolate2d alignCorners tensor    
+  | symbolVal (Proxy @mode) == "nearest" = unsafePerformIO $ (ATen.cast2 ATen.Managed.upsample_nearest2d_tl) tensor ([h, w] :: [Int])
```

Comment:
Did you forget to compute `newShape` here? It seems it's completely undetermined.

Can I also ask you to not use a `Symbol` for case splitting here? There are two alternatives that I find more appropriate: 
* split the function into two and name them appropriately.
* use a sum type, e.g. `Interpolate2dMode = Interpolate2dNearest | Interpolate2dBilinear` and add it to the arguments.

can you add a sum type for `alignCorners`? e.g. `Interpolate2dAlignCorners = Interpolate2dWithAlignedCorners | Interpolate2dWithoutAlignedCorners`